### PR TITLE
Add advisory for h2: resource exhaustion vulnerability may lead to DoS

### PR DESCRIPTION
**Pull request by KisaragiEffective**
(should not be merged as is at writing this comment, see following comment)

### CODE REVIEW

File: crates/h2/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+categories = ["denial-of-service"]
+keywords = ["http", "http2", "h2"]
+aliases = ["CVE-2023-26964", "GHSA-f8vr-r385-rh5r"]
+[versions]
+patched = [">= 0.3.17"]
+```
+
+# h2 causes stream stacking, may cause Denial of Service (DoS)
+
+h2 causes stream stacking, may cause Denial of Service.
```
Comment (tarcieri) [inline crates/h2/RUSTSEC-0000-0000.md:16]: The description in the [GHSA advisory was a bit more extensive](https://github.com/advisories/GHSA-f8vr-r385-rh5r).

Comment (tarcieri) [reply crates/h2/RUSTSEC-0000-0000.md]: I'm not saying to verbatim copy it, but it would be nice to have the same information

Comment (KisaragiEffective) [reply crates/h2/RUSTSEC-0000-0000.md]: I don't believe I can give appropriate summary for this entry.

Comment (tarcieri) [reply crates/h2/RUSTSEC-0000-0000.md]: Okay, well the added info is an improvement at least

**Comment by Noah-Kennedy** [comment]
Let's get @seanmonstar's sign off on this to make sure the wording is correct regarding what the vuln is.

The original CVE was not very descriptive of the issue, and mentioned only an implementation detail which of itself isn't necessarily exploitable. It was only in conjunction with other things that there could be an issue, we found.

File: crates/h2/RUSTSEC-0000-0000.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+categories = ["denial-of-service"]
+keywords = ["http", "http2", "h2"]
+aliases = ["CVE-2023-26964", "GHSA-f8vr-r385-rh5r"]
+[versions]
+patched = [">= 0.3.17"]
+```
+
+# h2 causes stream stacking, may cause Denial of Service (DoS)
+
+h2 causes stream stacking, may cause Denial of Service. Please see [GitHub Security Advisory entry](https://github.com/advisories/GHSA-f8vr-r385-rh5r) for more info.
```
Comment (seanmonstar) [inline crates/h2/RUSTSEC-0000-0000.md:16]: I'd never heard of stream stacking, and a quick Google search only brings up a copy of the same CVE in all the registries (😅).

```suggestion
If an attacker is able to flood the network with pairs of `HEADERS`/`RST_STREAM` frames, such that the `h2` application is not able to accept them faster than the bytes are received, the pending accept queue can grow in memory usage. Being able to do this consistently can result in excessive memory use, and eventually trigger Out Of Memory.

Please see the [GitHub Security Advisory entry](https://github.com/advisories/GHSA-f8vr-r385-rh5r) for more info.
```

Fair?

**Comment by Noah-Kennedy** [comment]
It might actually be better for him or myself to write this. This is a good place for us to pull the useful information out of our heads and put it to paper.

The original CVE wasn't helpful for maintainers or users, and I'd really like this one to be better.

**Comment by KisaragiEffective** [comment]
I believe so, I don't understand whole point at currenet moment nor have read entire thread around it.

File: crates/h2/RUSTSEC-0000-0000.md (hunk 2 above)
Comment (Noah-Kennedy) [reply crates/h2/RUSTSEC-0000-0000.md:16]: This is somewhat of an unusual case given the circumstances of that CVE.

Comment (seanmonstar) [reply crates/h2/RUSTSEC-0000-0000.md:16]: At the same time, we're guessing. The reporter hasn't confirmed that was the issue _they_ saw. So, I dunno 🤷

Comment (Noah-Kennedy) [reply crates/h2/RUSTSEC-0000-0000.md:16]: I believe the reporter of the issue was a different person than the reporter of the CVE. I think the CVE was just a copy-paste of the issue.

**Comment by tarcieri** [comment]
@seanmonstar can you confirm the advisory looks good now?

**Comment by seanmonstar** [comment]
LGTM! Thanks for all the work to keep this db running!

File: crates/h2/RUSTSEC-0000-0000.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+[versions]
+patched = [">= 0.3.17"]
+```
+
+# Resource exhaustion vulnerability in h2 may lead to Denial of Service (DoS)
+
+If an attacker is able to flood the network with pairs of `HEADERS`/`RST_STREAM` frames, such that the `h2` application is not able to accept them faster than the bytes are received, the pending accept queue can grow in memory usage. Being able to do this consistently can result in excessive memory use, and eventually trigger Out Of Memory.
+
+Please see the [GitHub Security Advisory entry](https://github.com/advisories/GHSA-f8vr-r385-rh5r) for more info.
+
```
Comment (tarcieri) [inline crates/h2/RUSTSEC-0000-0000.md:19]: This advisory now contains more detailed and correct information than the GHSA, so we probably shouldn't refer people to that.

```suggestion
Please see the [GitHub Security Advisory entry](https://github.com/advisories/GHSA-f8vr-r385-rh5r) for more info.

```

**Pull request merged**